SeamCarver.js

```diff
@@ -1,18 +1,33 @@
 "use strict";
 
+const RED = 0;
+const GREEN = 1;
+const BLUE = 2;
+const BORDER_ENERGY = 1000;
+
 class SeamCarver {
     constructor(canvas) {
         this.width = canvas.width;
         this.height = canvas.height;
         this.picture = [];
-        var context = canvas.getContext("2d");
+        this.context = canvas.getContext("2d");
         for (var y = 0; y < this.height; y ++) {
             for (var x = 0; x < this.width; x ++) {
-                var color = context.getImageData(x, y, 1, 1).data;
+                var color = this.context.getImageData(x, y, 1, 1).data;
                 var rgb = this.rgbToNum(color[0], color[1], color[2]);
                 this.picture[this.pixelToIndex(x, y)] = rgb;
             }
         }
+
+        // Simple implementation of energy matrix as array of arrays.
+        // Because we need to remove items, when removing the seam,
+        // maybe some sort of linked structure is more efficient.
+        this.energy_matrix = new Array(this.width);
```

Comment:
A single dimension array does remove one indirection. And calculating adjacent cells is as efficient as it gets: a simple multiplication and sum. But that does not solve the inefficiency of removing positions when removing a seam ...